client/src/components/EventForm/index.jsx

```diff
@@ -111,23 +161,84 @@ class EventForm extends Component {
       eventData.tags = tag_arr.map(obj => obj.tag);
     };
 
-    request('/event', 'POST', JSON.stringify(eventData))
-    .then(data => {
-      if ('errors' in data) {
-        for (let i = 0; i < data['errors'].length; i++) {
-          window.Materialize.toast(`${data.errors[i].name}: ${data.errors[i].description}`, 5000);
-          errorMessages[data.errors[i].name] = data.errors[i].description;
+    request(`/event/${this.props.match.params.id}`, 'PUT', JSON.stringify(eventData))
+      .then(data => {
+        if ('errors' in data) {
+          for (let i = 0; i < data['errors'].length; i++) {
+            window.Materialize.toast(`${data.errors[i].name}: ${data.errors[i].description}`, 5000);
+            errorMessages[data.errors[i].name] = data.errors[i].description;
+          };
+
+          this.setState({
+            error: {
+              ...this.state.error,
+              ...errorMessages,
+            }
+          });
+          return 1;
         };
-        
-        this.setState({error: {
-          ...this.state.error,
-          ...errorMessages,
-        }});
+
+        this.props.history.push(`/event/${this.props.match.params.id}`);
+      });
+  }
+
+  addEvent = (e) => {
+    let errorMessages = this.validateData();
+    if (!errorMessages) return 0;
+
+    let eventData = {
+      name: this.state.title,
+      lat: this.state.lat,
+      long: this.state.long,
+      description: this.state.description,
+      category: this.state.category,
+      start_date: moment(
+        [this.state.start_date, this.state.start_time].join(' '),
+        'D MMMM, YYYY h:mmA'
+      )._d.toJSON(),
+    };
+
+    if (this.state.end_date && this.state.end_time) {
+      eventData.end_date = moment(
+        [this.state.end_date, this.state.end_time].join(' '),
```

Comment:
This is add finished events to history in database or just confirm finish of the event?

Reply:
... sorry but I don't understand.
This part of the code gets date end time from the state (if they exist), converts to JSON valid format and saves in `eventData` object